src/util/http_utils.js

```diff
@@ -702,6 +702,15 @@ function validate_server_ip_whitelist(req) {
     throw new S3Error(S3Error.AccessDenied);
 }
 
+function validate_bucket_name_object_key_length(req) {
```

Comment:
I don't think this fits http_utils, it can just be in s3_rest, or if we need to share it then maybe in s3_utils